packages/graph-engine/src/programmatic/input.ts

```diff
@@ -59,7 +59,27 @@ export class Input<T = any> extends Port<T> {
       }
       //Otherwise do a structural comparison
       else if (JSON.stringify(this._type) !== JSON.stringify(opts.type)) {
-        this._dynamicType = opts.type;
+        const {variadic, _dynamicType } = this;
+
+        // for variadic ports, we need to set the dynamic type depending on all the other connected types
+        if (variadic && _dynamicType && JSON.stringify(_dynamicType) !== JSON.stringify(opts.type)) {          
+          const graph = this.node.getGraph();
+          const inEdges = graph.inEdges(this.node.id);
+          const sourceNodesOutputs = inEdges.map(edge => edge.source).map(id => graph.getNode(id).outputs);
+          const sourceNodesOuputTypes = sourceNodesOutputs.map(output => output.value?.type);
+
+          if (sourceNodesOuputTypes.every(type => type?.$id === sourceNodesOuputTypes[0].$id)) {
+              this._dynamicType = { 
+                type: 'array',
+                items: sourceNodesOuputTypes[0]
+              }
+          } else {
+            this._dynamicType = AnyArraySchema;
```

Comment:
This looks like it would undo the type which could break some json schemas. We should probably just remove the dynamic type in that case instead of giving the any type

Reply:
ok

Reply:
But this was te requirement. when we connect multiple types, the dynamic type should be any

Reply:
Yeah but this this gives no additional info. Likely the underlying type is either already an any array or a permissive type. Unless all of the values are the same adding any array as a dynamic type will just complicate debugging